src/main/java/org/jenkinsci/plugins/scm_filter/GitHubPullRequestSkipTrait.java

```diff
@@ -0,0 +1,106 @@
+package org.jenkinsci.plugins.scm_filter;
+
+import hudson.Extension;
+import hudson.plugins.git.GitSCM;
+import hudson.scm.SCMDescriptor;
+
+import jenkins.scm.api.SCMHead;
+import jenkins.scm.api.trait.*;
+
+import org.jenkinsci.plugins.github_branch_source.GitHubSCMBuilder;
+import org.jenkinsci.plugins.github_branch_source.GitHubSCMSourceRequest;
+import org.jenkinsci.plugins.github_branch_source.PullRequestSCMHead;
+
+import org.kohsuke.github.GHRepository;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+
+
+public class GitHubPullRequestSkipTrait extends SCMSourceTrait {
+
+
+    @DataBoundConstructor
+    public GitHubPullRequestSkipTrait () {
+    }
+
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void decorateContext (SCMSourceContext<?, ?> context) {
+        context.withFilter(new GitHubPullRequestSkipTrait.ExcludePRsSCMHeadFilter());
+    }
+
+
+    /**
+     * Our descriptor.
+     */
+    @Extension
+    @SuppressWarnings("unused") // instantiated by Jenkins
+    public static class PullRequestSkipTraitDescriptorImpl extends SCMSourceTraitDescriptor {
+
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        @Nonnull
+        public String getDisplayName () {
+            return "Pull request title filtering behaviour";
+        }
+
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public boolean isApplicableToSCM (@Nonnull SCMDescriptor<?> scm) {
+            return scm instanceof GitSCM.DescriptorImpl;
+        }
+
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public boolean isApplicableToBuilder (@Nonnull Class<? extends SCMBuilder> builderClass) {
+            return GitHubSCMBuilder.class.isAssignableFrom(builderClass);
+        }
+    }
+
+
+    /**
+     * Filter that excludes pull requests according to its PR title
+     * if it contains [ci skip], [skip ci], or [wip], case insensitive.
+     * The words "ci" and "skip" can be separated by either a dash, an underscore, or a single space.
+     */
+    public static class ExcludePRsSCMHeadFilter extends SCMHeadFilter {
+
+
+        ExcludePRsSCMHeadFilter () {
+        }
+
+
+        @Override
+        public boolean isExcluded (
+                @Nonnull SCMSourceRequest scmSourceRequest,
+                @Nonnull SCMHead scmHead
+        ) throws IOException, InterruptedException {
+
+            if (scmHead instanceof PullRequestSCMHead) {
+                GHRepository repository = ((GitHubSCMSourceRequest) scmSourceRequest).getRepository();
+
+                // name for pull requests are formatted as "PR-<number>"
+                int prNumber = Integer.parseInt(scmHead.getName().substring(3));
+
+                String prTitle = repository.getPullRequest(prNumber).getTitle();
+                return prTitle.matches("(?i).*\\[(wip|ci[ -_]skip|skip[ -_]ci)\\].*");
```

Comment:
This means every time this method is executed, the regex is recompiled. Is it possible to assign this in a static variable?
```java
public static class ExcludePRsSCMHeadFilter extends SCMHeadFilter {

    private static Pattern skipSymbol = Pattern.compile("\\[(wip|ci[ -_]skip|skip[ -_]ci)\\]", Pattern.CASE_INSENSITIVE);
    ...
    public boolean isExcluded(...) {
    // find is faster than matches as it does not exhaust the pattern, we only need a match for the skipSymbol not the whole string containing the symbol
        return ExcludePRsSCMHeadFilter.skipSymbol.matcher(prTitle).find();
    }
}
```

Reply:
I think the current implementation is better because it is simpler. We don't really need the operation to be very fast, and the benefit will be very minimal.

Wdyt?

Reply:
This is actually been implemented before and I don't even know the reason why you changed it...
Is it readability?
Is `(?i)` symbol more readable than a class constant `Pattern.CASE_INSENSITIVE`? 
I'm just not sure why does this implementation keep changing. Also, if you're not concerned with the optimization, at least don't worsen it, if the previous implementation is also already readable.

Reply:
It was changed to become simpler.
* Less package needs to be imported
* Less variables are used
* The speed is still very acceptable

Although it is less optimal than your suggestion, the speed does not really suffer. Especially for a function that only gets called once an hour/every time someone pushes a commit.

Reply:
I agree performance is not really a concern for this implementation atm.